core/datatrak.c

```diff
@@ -639,7 +639,9 @@ bool dt_dive_parser(FILE *archivo, struct dive *dt_dive)
 		 * 2bytes per sample plus another one each three samples. Also includes the
 		 * bytes jumped over (22) and the nitrox (2) or O2 (3).
 		 */
-		int samplenum = is_O2 ? (profile_length - 25) * 3 / 8 : (profile_length - 24) * 3 / 7;
+		int numerator = is_O2 ? (profile_length - 25) * 3 : (profile_length - 24) * 3;
+		int denominator = is_O2 ? 8 : 7;
+		int samplenum = (numerator / denominator) + (((numerator % denominator) != 0) ? 1 : 0);
```

Comment:
Good night.  There are two issues here:
1.-- For some dives we are missing some samples, not just one, but may be 6 to 9, even 12 (up to 3 minutes or more).   I need to look into the dives to find a pattern, as maths for this are easy  (n-24)÷(2+(1÷3)) for a nitrox device, where n is the readed profile length in bytes.
2.-- After ending a dive parsing, we search for a pattern A0 00 that marks the beginning of another dive.  This is done looking  for the A0 byte and checking the next to be 00.  The code for this used to work well, but actual one is completely bogus, and I can't remember when did this mess. Will send tomorrow a PR to fix this with a hackish and dirty workaround.

In the end the full importer has to be reworked because we don't really need to read the file, it has been read in parse_file() and put in a memblock struct, so we just need to parse the buffer using some pointer maths.

BTW I had been valuing in the past just passing the full profile to libdivecomputer, in the way we do in smarttrak import, and let libdc do the hard work.

As said a complete rework ...